config/site/src/_includes/filtering_predicate_definitions/fulltext.md

```diff
@@ -3,11 +3,11 @@
   full text search. This is stricter than `matchesQuery`: all terms must match
   and be in the same order as the provided phrase.
 
-  Will be ignored when `null` is passed.
+  Will be ignored when `null` is passed as the `MatchesPhraseFilterInput`.
```

Comment:
I'm on the fence about including the changes to this file, for a couple reasons:

* This is copied verbatim from the GraphQL docs for the `TextInputFilter.matchesPhrase` and `TextInputFilter.matchesQuery` fields. It'd be nice to not diverge from them. If this is needed, we should probably update the GraphQL docs, too.
* The wording `Will be ignored when null is passed` is used consistently throughout our GraphQL docs (as shown below it shows up 148 times in our `.graphql` [schema file](https://github.com/block/elasticgraph/blob/main/config/schema/artifacts/schema.graphql)!). If we're going to update this, should we update all of them with a similar clarification?
* Does adding this note about the type bring necessary clarity? "Will be ignored when `null` is passed" is intended to apply to the input field that's being documented. In this case, the `matchesPhrase` field--it indicates that `matchesPhrase: null` will cause it to be ignored.

Thoughts?

<img width="428" alt="image" src="https://github.com/user-attachments/assets/7d5c5c25-94a3-4a05-a135-50f866f9434d">

Reply:
That makes sense. I've reverted the change and can see where you're coming from.

The confusion on my side stemmed from the fact that most of the examples show how to populate the `matchesQuery` parameters, so I was assuming passing `null` to the `query` parameter of `matchesQuery` would cause the filter to be ignored; however, it led to a very vague error about the type being unexpected... So maybe there's something we can do to improve that error, instead? WDYT?

Reply:
> however, it led to a very vague error about the type being unexpected...

The type error is the equivalent of a compiler error in a statically typed language such as when the type of an argument doesn't align with the type of the method signature.  We're not in control of it (it's provided by the graphql gem and might even be strictly specified by the GraphQL spec), and it's worth noting that it's not unique to this part of the ElasticGraph query API. You'd get a similar error anytime you write a query that attempts to pass a `null` value for a non-null input field.

I think it's part of the learning curve of GraphQL.